framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -1315,6 +1315,10 @@ protected void addJoins(StringBuilder str, Collection<JoinBuilder<SearchCriteria
     public Pair<List<T>, Integer> searchAndCount(final SearchCriteria<T> sc, final Filter filter) {
         List<T> objects = search(sc, filter, null, false);
         Integer count = getCount(sc);
+        // Count cannot be less than the result set but can be higher due to pagination, see CLOUDSTACK-10320
+        if (count < objects.size()) {
```

Comment:
This logic may be used in changes below.

Reply:
I will change how it's handled